libs/networking/WordPressNetworking/src/main/java/org/wordpress/android/networking/RestClientUtils.java

```diff
@@ -196,47 +198,45 @@ public void getCurrentTheme(long siteId, Listener listener, ErrorListener errorL
+    public void getGeneralSettings(long siteId, Listener listener, ErrorListener errorListener) {
         String path = String.format(Locale.US, "sites/%d/settings", siteId);
-        post(path, params, null, listener, errorListener);
-    }
-
-    public void getJetpackSettings(long siteId, Listener listener, ErrorListener errorListener) {
-        String path = String.format(Locale.US, "jetpack-blogs/%d", siteId);
         get(path, listener, errorListener);
     }
 
-    public void getJetpackProtect(long siteId, Listener listener, ErrorListener errorListener) {
-        String path = String.format(Locale.US, "sites/%d/jetpack/modules/protect", siteId);
+    public void getJetpackMonitorSettings(long siteId, Listener listener, ErrorListener errorListener) {
+        String path = String.format(Locale.US, "jetpack-blogs/%d", siteId);
         get(path, listener, errorListener);
     }
 
-    public void setJetpackProtect(long siteId, boolean enabled, Listener listener, ErrorListener errorListener) {
-        String path = String.format(Locale.US, "sites/%d/jetpack/modules/protect", siteId);
-        Map<String, String> params = new HashMap<>();
-        params.put("active", String.valueOf(enabled));
+    public void setGeneralSiteSettings(long siteId, JSONObject params, Listener listener, ErrorListener errorListener) {
+        String path = String.format(Locale.US, "sites/%d/settings", siteId);
         post(path, params, null, listener, errorListener);
     }
 
-    public void getJetpackMonitor(long siteId, Listener listener, ErrorListener errorListener) {
-        String path = String.format(Locale.US, "sites/%d/jetpack/modules/monitor", siteId);
-        get(path, listener, errorListener);
-    }
-
-    public void setJetpackMonitor(long siteId, boolean enabled, Listener listener, ErrorListener errorListener) {
-        String path = String.format(Locale.US, "sites/%d/jetpack/modules/monitor", siteId);
-        Map<String, String> params = new HashMap<>();
-        params.put("active", String.valueOf(enabled));
-        post(path, params, null, listener, errorListener);
+    public void setJetpackSettings(long siteId, Map<String, Object> bodyData,
+                                  Listener listener, ErrorListener errorListener) {
+        String path = String.format(Locale.US, "jetpack-blogs/%d/rest-api/", siteId);
+        JSONObject params = new JSONObject();
+        JSONObject body = new JSONObject();
+        try {
+            for (String key : bodyData.keySet()) {
+                body.putOpt(key, bodyData.get(key));
+            }
+            params.put("path", "/jetpack/v4/settings/");
+            params.put("body", body.toString());
+            post(path, params, null, listener, errorListener);
+        } catch (JSONException e) {
+            AppLog.e(AppLog.T.API, "Error updating Jetpack settings: " + e);
```

Comment:
I think I missed this in my previous review. We should call the error listener here.